base_serializer.go

```diff
@@ -211,22 +212,37 @@ func (baseSerializer *baseSerializer) getBytes(ionType ion.Type, ionValue interf
 	} else if ionType == ion.FloatType && ionValue == 0 && int64(ionValue.(float64)) >= 0 {
 		// value is 0.0, not -0.0
 		return []byte{0x40}, nil
-	} else {
-		buf := bytes.Buffer{}
-		writer := ion.NewBinaryWriter(&buf)
+	}
 
-		err := serializers(ionType, ionValue, writer)
-		if err != nil {
-			return nil, err
-		}
+	buf := bytes.Buffer{}
+	writer := ion.NewBinaryWriter(&buf)
 
-		err = writer.Finish()
-		if err != nil {
-			return nil, err
-		}
+	err := serializers(ionType, ionValue, writer)
+	if err != nil {
+		return nil, err
+	}
 
-		return buf.Bytes()[4:], nil
+	err = writer.Finish()
+	if err != nil {
+		return nil, err
 	}
+
+	bytes := buf.Bytes()[4:]
+
+	if ionType == ion.FloatType && len(bytes) == 5 {
+		// We got back a float32 but we want to hash it as a float64.
```

Comment:
I'd suggest making a reference to the Ion spec here: https://amzn.github.io/ion-hash/docs/spec.html#4-float

Reply:
Done.